test-data/unit/check-overloading.test

```diff
@@ -2937,3 +2937,416 @@ def add_proxy(x, y):
 tup = (1, '2')
 reveal_type(foo(lambda (x, y): add_proxy(x, y), tup))  # E: Revealed type is 'builtins.str*'
 [builtins fixtures/primitives.pyi]
+
+[case testOverloadWithClassMethods]
+from typing import overload
+
+class Wrapper:
+    @overload
+    @classmethod
+    def foo(cls, x: int) -> int: ...
+    @overload
+    @classmethod
```

Comment:
One more suggestion: could you please add to your TODO list adding few tests to check that overloads work well with self-types (both instance, and class methods using `cls: Type[T]`)?